### ISSUE TITLE
Drag&Drop entry (WWW) to a group

### ISSUE THREAD

**Issue by noMICROSOFTbuhtz** [JabRef/jabref#4664]
JabRef 5.0-dev--snapshot--2019-02-19--master--623718aa9
Linux 4.9.0-8-amd64 amd64 
Java 1.8.0_181

When I drag&drop an entry (type WWW) to group nothing happens. The entry does not appear in that group. But the entry itself updates it "groups" field with the name of the new group. Restart doesn't help.

I **can not** reproduce this for all entries but for this.

Please look in the group **_entry_date -> 2019-02** and try to drag the entry in it to the group **Science -> Statistic**. Also look into the `groups` field of the entry!
[MyJabRef_bug_report.bib.zip](https://github.com/JabRef/jabref/files/2879608/MyJabRef_bug_report.bib.zip)

**Comment by Siedlerchr** [comment]
Works fine the latest master:
![grafik](https://user-images.githubusercontent.com/320228/54474679-17df3680-47e8-11e9-83e5-457ae3d3af45.png)

**Comment by noMICROSOFTbuhtz** [comment]
You did not used the bib-file uploaded by me. Please test it with this and follow my step-by-step to reproduce. I can still reproduce the problem with the dev from today.

**Comment by AEgit** [comment]
Cannot confirm for (@noMICROSOFTbuhtz : I have used the bib file you provided):

JabRef 5.0-dev--snapshot--2019-03-20--master--6c61664a8
Windows 10 10.0 amd64 
Java 1.8.0_201

So if the issue still appears with your setup, it might be configuration- or Linux-specific.

Or do you just mean, that the article count is not updated in the group panel? If that is the case - that issue is already known, see here:

https://github.com/JabRef/jabref/issues/4441
https://github.com/JabRef/jabref/issues/4679

**Comment by noMICROSOFTbuhtz** [comment]
I will do some more tests on other systems and will report back. Please give me some time.

**Comment by Siedlerchr** [comment]
Okay, I looked into it and it seem it's a view configuration problem of the group:
Your statistic group is defined as "Refine supergroup" 
![grafik](https://user-images.githubusercontent.com/320228/54951190-4a550600-4f43-11e9-80f5-b969036afb43.png)
If you choose Independent group it's displayed correctly.  

As I'm not an expert of the groups mode stuff, maybe @AEgit  can help clarify the group modes

**Comment by noMICROSOFTbuhtz** [comment]
I can reproduce this here. Thanks for the hint.

**Comment by AEgit** [comment]
I have to take it back, I can reproduce the behaviour and I think this is a bug:

JabRef 5.0-dev--snapshot--2019-03-28--master--a73167933
Windows 10 10.0 amd64 
Java 1.8.0_201

As @Siedlerchr mentioned, the issue only appears when "Refine supergroup" is selected. I generally do not use this feature but use "Independent group", which does not display the problem.

However, as far as I understand the "Refine supergroup" setting is meant to contain BOTH the subgroup entries AND the entries in the subgroup, i. e. following a Boolean "OR" statement (see http://help.jabref.org/en/Groups). But even, if a Boolean "AND" is intended, the issue still persists - the entry is not shown in the maintable, despite having selected the subgroup of interest.

I have tried this also with my own database and it appears to be a general problem with the "Refine supergroup" setting. So, apologies to @noMICROSOFTbuhtz for initially not being able to reproduce this behaviour and thanks to @Siedlerchr for spotting the problem.

**Comment by tobiasdiez** [comment]
The point of the "refine supergroup" is that it shows only entries that are matched by the supergroup and in addition also by the group itself. To take the example from the doc: `distribution > gauss`, the `gauss` group only shows entries that have the keywords `distribution` and `gauss`.

**Comment by Siedlerchr** [comment]
So refine supergroup only works for keyword based groups and not for static groups with manual assigned entries? 

From the definition I though it shows all entries in the current child group B and all from the supergroup A. 
So the logical opposite of selecting group A and showing entries from the child groups B and C

**Comment by tobiasdiez** [comment]
That's how it is supposed to work:
https://github.com/JabRef/jabref/blob/8777c16a99bd83e4a30ddafc859c0c2208b68733/src/main/java/org/jabref/model/groups/GroupHierarchyType.java#L5-L18

**Comment by AEgit** [comment]
Ok, if that is the case, the "Refine Supergroup" function appears to work as intended - initially I thought that irrespective of the interpretation it did not work since I assigned the item mentioned in the starting post to both the group "Statistic" and "Science" in the provided bib file. The item would then be shown in "Science" but not in "Statistic" - but that makes now sense, since "Statistic" is NOT the immediate subgroup of "Science" but that of "Methode". If I assign the item also to "Methode" it will now also appear in "Statistic".
So apologies again for the confusion! Maybe the only take-home message here is that the help file needs to be a bit more clear about this. I have added a single sentence to explain the feature a little bit better, see here:
https://github.com/JabRef/help.jabref.org/pull/202
